myapp/app/controllers/tasks_controller.rb

```diff
@@ -1,6 +1,21 @@
+require 'uri'
+
 # TaskController is a controller to handle basic CRUD operations for "task"
 class TasksController < ApplicationController
+  include ApplicationHelper
+
   def index
+    q = Task
+
+    unless params[:query].to_s.empty?
+      # TODO: fix this. this fuzzy search might cause performance degradation.
+      q = q.where('title LIKE ?', "%#{params[:query]}%")
```

Comment:
It looks this code is vulnerable to SQL injection attacks. 
https://techracho.bpsinc.jp/hachi8833/2021_12_06/114096

Reply:
Thanks for the suggestion. I fixed and pushed it. It is good to know that prepared statement of Active Record isn't enough to escape for wildcard chars.

I tested several cases on my local to check how param is escaped.

search term
**%') OR 1 = 1; -- %**

```
# raw query
q = q.where("title LIKE '%#{params[:query].to_s}%'")
api-1     |   Task Count (0.3ms)  SELECT COUNT(*) FROM `tasks` WHERE (title LIKE '%%') OR 1 = 1; -- %%')

# prepared statement
q = q.where('title LIKE ?', "%#{params[:query].to_s}%")
api-1     |   Task Count (0.2ms)  SELECT COUNT(*) FROM `tasks` WHERE (title LIKE '%%\') OR 1 = 1; -- %%')

# using sanitize function
q = q.where('title LIKE ?', "%#{ActiveRecord::Base.sanitize_sql_like(params[:query].to_s)}%")
api-1     |   Task Count (0.2ms)  SELECT COUNT(*) FROM `tasks` WHERE (title LIKE '%\\%\') OR 1 = 1; -- \\%%')
```